core/tx_multicurrency_priceheap_test.go

```diff
@@ -320,7 +328,47 @@ func TestIsCheaper(t *testing.T) {
 	var notCheaper CurrencyCmpFn = func(p1 *big.Int, c1 *common.Address, p2 *big.Int, c2 *common.Address) int {
 		return 1
 	}
-	assert.True(t, cheaper.IsCheaper(tx1, tx2))
-	assert.False(t, equal.IsCheaper(tx1, tx2))
-	assert.False(t, notCheaper.IsCheaper(tx1, tx2))
+	assert.True(t, cheaper.Cmp(tx1, tx2, nil, nil) <= 0)
+	assert.True(t, equal.Cmp(tx1, tx2, nil, nil) <= 0)
+	assert.False(t, notCheaper.Cmp(tx1, tx2, nil, nil) <= 0)
+}
+
+// TestMulticurrencyUnderpriced tests that the underpriced method from pricedList functions
+// properly when handling many different currencies.
+func TestMulticurrencyUnderpriced(t *testing.T) {
+	all := newTxLookup()
+	curr1 := common.HexToAddress("curr1")
+	rate1, _ := currency.NewExchangeRate(common.Big1, common.Big1)
+	curr2 := common.HexToAddress("curr2")
+	rate2, _ := currency.NewExchangeRate(common.Big1, common.Big1)
```

Comment:
It's actually results in a single currency as `common.HexToAddress("curr1")` and `commont.HexToAddress("curr2")` both would return `0x000000000000000000000000000000000000000C` as neither `curr1` nor `curr2` is a valid hex string.
Also, it would make sense to use something less trivial than exchange rate 1.